### ISSUE TITLE
AST_COMPARE Troubleshooting

### ISSUE THREAD

**Issue by ionrocks** [prettier/plugin-php#514]
A few days back, I inherited an older code base that I really wanted to make prettier before diving in! I found this plugin and thought I'd give it a try (despite the alpha status), but unfortunately there's a few parts of the code base that are tripping it up. I dove in to the code to see if I could figure out what was going on, but I hit a snag early on with the tests not seeming to run correctly!

Steps To Reproduce:

```bash
git clone https://github.com/prettier/plugin-php.git
cd plugin-php
yarn
yarn test
AST_COMPARE=1 yarn test
```

With and without the `AST_COMPARE=1` setting, the result is the same: all tests pass. I don't think this is correct, even for some of the existing test cases!

Long story short, some logging showed that the comparison wasn't even running, so I made some changes, got it working for me, and pushed that code here:

https://github.com/ionrocks/plugin-php/tree/testing-updates

I didn't make a pull request because I don't know this code base very well, and am hoping someone will tell me if I'm missing something obvious. 😃

**Comment by czosel** [comment]
Hmm, if I'm not mistaken we _do_ run the AST comparison, but we accept some differences between the original and the formatted AST. In prettier core this "cleanup" stage is called `massageAstNode`, and we implented here: https://github.com/prettier/plugin-php/blob/master/src/clean.js

**Comment by evilebottnawi** [comment]
@czosel what do you think about this? How we can solve this?
